File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums) {
        //let len = nums.length;
        let res = [];
        for(let i = 0; i < nums.length; i++){
           let re = 1;
            for(let j = 0; j < nums.length; j++){
                if(j !== i){
                    re *=  nums[j];
                }
            }
            res.push(re);
        }
        return res;
    }
}

File: Data Structures & Algorithms/permutations/submission-1.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        res = []
        def rec(arr,ind):
            if ind == len(arr):
                res.append(arr[:])
                return
            for i in range(ind,len(arr)):
                    arr[i],arr[ind] = arr[ind],arr[i]
                    rec(arr,ind+1)
                    arr[i],arr[ind] = arr[ind],arr[i]
        rec(nums,0)
        return(res)